File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ManaBoots.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class ManaBoots extends Equippable {
	{
		name = Locale.get("mana_boots");
		description = Locale.get("mana_boots_desc");
		sprite = "item-running_boosts_mana_regen";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).flipRegenFormula = true;
		}
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).flipRegenFormula = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ManaBoots.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class ManaBoots : Equippable() {
	init {
		name = Locale.get("mana_boots")
		description = Locale.get("mana_boots_desc")
		sprite = "item-running_boosts_mana_regen"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.flipRegenFormula = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.flipRegenFormula = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}